pandas/tests/frame/test_constructors.py

```diff
@@ -32,12 +32,33 @@
 
 class TestDataFrameConstructors(TestData):
 
-    def test_constructor(self):
-        df = DataFrame()
-        assert len(df.index) == 0
-
-        df = DataFrame(data={})
-        assert len(df.index) == 0
+    @pytest.mark.parametrize('constructor', [
+        lambda: DataFrame(),
+        lambda: DataFrame(None),
+        lambda: DataFrame({}),
+        lambda: DataFrame(()),
+        lambda: DataFrame([]),
+        lambda: DataFrame((x for x in [])),
+        lambda: DataFrame(data=None),
+        lambda: DataFrame(data={}),
+        lambda: DataFrame(data=()),
+        lambda: DataFrame(data=[]),
+        lambda: DataFrame(data=(x for x in [])),
+        # these are NOT empty DataFrames
+        pytest.param(lambda: DataFrame([[]]), marks=pytest.mark.xfail(
```

Comment:
Better to parametrize these in a separate test rather than supplying mark here

Reply:
I was just cautious of making actual behaviour tested and then tested behaviour becoming the accepted behaviour because it's tested. I'm not sure if the index differences are intentional.

Reply:
yeah i agree, can you separate these cases out

Reply:
can you update this and merge master, ping on green.